src/workers/sendAnalyticsReportMail.ts

```diff
@@ -19,40 +19,58 @@ const worker: Worker = {
     const data: Data = messageToJson(message);
     try {
       const post = await con.getRepository(Post).findOne(data.postId);
-      if (!post.sentAnalyticsReport && post.authorId) {
-        const user = await fetchUser(post.authorId);
-        const source = await post.source;
-        const stats = await getAuthorPostStats(con, user.id);
-        await sendEmail({
-          ...baseNotificationEmailData,
-          to: user.email,
-          templateId: templateId.analyticsReport,
-          dynamicTemplateData: {
-            first_name: user.name.split(' ')[0],
-            source_image: source.image,
-            post_image: post.image || pickImageUrl(post),
-            post_title: truncatePostToTweet(post),
-            live_hours: differenceInHours(new Date(), post.createdAt),
-            post_views: post.views?.toLocaleString() ?? 0,
-            post_views_total: stats.numPostViews?.toLocaleString() ?? 0,
-            post_upvotes: post.upvotes?.toLocaleString() ?? 0,
-            post_upvotes_total: stats.numPostUpvotes?.toLocaleString() ?? 0,
-            post_comments: post.comments?.toLocaleString() ?? 0,
-            user_reputation: user.reputation?.toLocaleString() ?? 0,
-            profile_image: user.image,
-            full_name: user.name,
-            user_handle: user.username,
-            profile_link: user.permalink,
-          },
-        });
-        logger.info(
-          {
-            data,
-            messageId: message.messageId,
-          },
-          'analytics report sent',
-        );
+      if (!post.authorId && !post.scoutId) {
```

Comment:
should we not add `post.sentAnalyticsReport` here? In case it's false?

Reply:
I agree, it would ensure we won't be sending any email if not necessary.